### ISSUE TITLE
W5500 "setHostname" issue

### ISSUE THREAD

**Issue by basilmozg** [sstaub/Ethernet3#51]
"setHostname" dont work for me
I use W5500 shield and ESO8266

Init
`Ethernet.setHostname("ethernet");
Ethernet.begin(mac);`

DHCP return address
But when I link for ethernet.local there os no answer in browser.
What am I doing wrong?
P.S. Wifi host name works fine

**Comment by sstaub** [comment]
Can try the code from the main branch?

**Comment by basilmozg** [comment]
Do you mean a code from examples? There are no examples with .setHostname
I have tried use `const *char host = "ethernet"` but still the same result.
Am I right when enter "hostname.local" in the browser?

**Comment by sstaub** [comment]
No, I mean the whole library itself, you can download it directly from the main branch and then install the library manually. There was a change in handling DHCP memory, maybe it solves your problem.

**Comment by basilmozg** [comment]
I sure this is the same last version. But anyway I reload it. Still the same negative result

` modified 12 Aug 2013
 by Soohwan Kim (suhwan@wiznet.co.kr)

 - 10 Apr. 2015
 Added support for Arduino Ethernet Shield 2
 by Arduino.org team

name=Ethernet3
version=1.5.5`

**Comment by MicSG-dev** [comment]
You are confusing hostname with mDNS.
To access http://ethernet.local/ you need to use an mDNS library. I usually use the [ArduinoMDNS](https://github.com/arduino-libraries/ArduinoMDNS) library. See the example [RegisteringServices.ino](https://github.com/arduino-libraries/ArduinoMDNS/blob/master/examples/Ethernet/RegisteringServices/RegisteringServices.ino).

The `Ethernet.setHostname("ethernet");` function only configures the hostname that will be used by the Ethernet interface when connecting to the network. This name is presented to the DHCP server. You can see this information in your router's interface. Here is my ESP32 with W5500 ethernet module listed in the TP-Link router interface:
![Image](https://github.com/user-attachments/assets/707d2f56-b3f1-4cc6-a53a-cc7cd7c73374)
If I used `Ethernet.setHostname("ethernet");` I would see the following in the router interface:

![Image](https://github.com/user-attachments/assets/eca9cc1c-eb8b-4464-84c2-91010d04e618)

I hope you understand the difference between hostname and mDNS.
